android-reactive-location/src/main/java/pl/charmas/android/reactivelocation/ReactiveLocationProvider.java

```diff
@@ -92,8 +93,39 @@ public ReactiveLocationProvider(Context ctx) {
      * @param maxResults maximal number of results you are interested in
      * @return observable that serves list of address based on location
      */
-    public Observable<List<Address>> getGeocodeObservable(double lat, double lng, int maxResults) {
-        return GeodecodeObservable.createObservable(ctx, lat, lng, maxResults);
+    public Observable<List<Address>> getReverseGeocodeObservable(double lat, double lng, int maxResults) {
+        return ReverseGeocodeObservable.createObservable(ctx, lat, lng, maxResults);
+    }
+
+    /**
+     * Creates observable that translates a street address or other description into a list of
+     * possible addresses using included Geocoder class. You should subscribe for this
+     * observable on I/O thread.
+     * The stream finishes after address list is available.
+     *
+     * @param locationName a user-supplied description of a location
+     * @param maxResults max number of results you are interested in
+     * @return observable that serves list of address based on location name
+     */
+    public Observable<List<Address>> getGeocodeObservable(String locationName, int maxResults) {
+        return getGeocodeObservable(locationName, maxResults, null);
+    }
+
+    /**
+     * Creates observable that translates a street address or other description into a list of
+     * possible addresses using included Geocoder class. You should subscribe for this
+     * observable on I/O thread.
+     * The stream finishes after address list is available.
+     * <p/>
+     * You may specify a bounding box for the search results.
+     *
+     * @param locationName a user-supplied description of a location
+     * @param maxResults max number of results you are interested in
+     * @param bounds restricts the results to geographical bounds
```

Comment:
Maybe we should document that bounds can be null not to force user to look into the original docs or source code?